examples/applications/plot_cyclical_feature_engineering.py

```diff
@@ -0,0 +1,741 @@
+#
+#     When reporting performance measure on the test set in the discussion, we
+#     instead choose to focus on the mean absolute error that is more
+#     intuitive than the (root) mean squared error. Note however that the best
+#     models for one metric are also the best for the other in this study.
+y = df["count"] / 1000
+_ = y.hist(bins=30)
+
+# %%
+# The input feature data frame is a time annotated hourly log of variables
+# describing the weather conditions. It includes both numerical and categorical
+# variables. Note that the time information has already been expanded into
+# several complementary columns.
+#
+X = df.drop("count", axis="columns")
+X
+# %%
+# We first introspect the distribution of the categorical variables, starting
+# with `"weather"`:
+#
+X["weather"].value_counts()
+
+# %%
+# Since there are only 3 `"heavy_rain"` events, we cannot use this category to
+# train machine learning models with cross validation. Instead, we simplify the
+# representation by collapsing those into the `"rain"` category.
+#
+X['weather'].replace(to_replace='heavy_rain', value="rain", inplace=True)
+# %%
+X["weather"].value_counts()
+
+# %%
+# As expected, the `"season"` variable is well balanced:
+#
+X["season"].value_counts()
+
+# %%
+# Time-based cross-validation
+# ---------------------------
+#
+# Since the dataset is a time-ordered event log (hourly demand), we will use a
+# time-sensitive cross-validation splitter to evaluate our demand forecasting
+# model as realistically as possible. We use a gap of 2 days between the train
+# and test side of the splits. We also limit the training set size to make the
+# performance of the CV folds more stable.
+#
+# 1000 test datapoints should be enough to quantify the performance of the
+# model. This represents a bit less than a month and a half of contiguous test
+# data:
+
+from sklearn.model_selection import TimeSeriesSplit
+
+ts_cv = TimeSeriesSplit(
+    n_splits=5, gap=48, max_train_size=10000, test_size=1000,
+)
+
+# %%
+# Let us manually inspect the various splits to check that the
+# `TimeSeriesSplit` works as we expect, starting with the first split:
+all_splits = list(ts_cv.split(X, y))
+train_0, test_0 = all_splits[0]
+
+# %%
+X.iloc[test_0]
+
+# %%
+X.iloc[train_0]
+
+# %%
+# We now inspect the last split:
+train_4, test_4 = all_splits[4]
+
+# %%
+X.iloc[test_4]
+
+# %%
+X.iloc[train_4]
+
+# %%
+# All is well. We are now ready to do some predictive modeling!
+#
+# Gradient Boosting
+# -----------------
+#
+# Gradient Boosting Regression with decision trees is often flexible enough to
+# efficiently handle heteorogenous tabular data with a mix of categorical and
+# numerical features as long as the number of samples is large enough.
+#
+# Here, we do minimal ordinal encoding for the categorical variables and then
+# let the model know that it should treat those as categorical variables by
+# using a dedicated tree splitting rule.
```

Comment:
It might be nice to mention that we explicitly provide the order of the categories to avoid automatic ordering based on lexicography.

Reply:
done.